synapse/storage/databases/main/appservice.py

```diff
@@ -378,7 +384,9 @@ async def get_new_events_for_appservice(
     ) -> Tuple[int, List[EventBase]]:
         """Get all new events for an appservice"""
 
-        def get_new_events_for_appservice_txn(txn):
+        def get_new_events_for_appservice_txn(
+            txn: LoggingTransaction,
+        ) -> Tuple[int, Collection[str]]:
```

Comment:
Any particular reason for returning a `Collection` rather than a `List`?

Reply:
No. I have changed it to `List`.

Reply:
Cheers! FWIW, my rule of thumb here is that `Collection` and the [other abstract types](https://docs.python.org/3/library/typing.html?highlight=typing#abstract-base-classes) are good for annotating function parameters. but I think return types are often best typed more literally.

(That is just an Opinion though...)